Doxygen/man/man3/TileSet.3
.TH "TileSet" 3 "Fri Jan 21 2022" "Version 1.0" "Neon Jumper" \" -*- nroff -*-
.ad l
.nh
.SH NAME
TileSet
.SH SYNOPSIS
.br
.PP
.PP
\fC#include <TileSet\&.h>\fP
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBTileSet\fP ()"
.br
.RI "Konstruktor pusty\&. "
.ti -1c
.RI "\fBTileSet\fP (int \fBfirstgid\fP, std::ifstream tileset)"
.br
.ti -1c
.RI "\fB~TileSet\fP ()"
.br
.RI "Destruktor\&. "
.ti -1c
.RI "int \fBgetFirstGid\fP ()"
.br
.ti -1c
.RI "int \fBgetColumns\fP ()"
.br
.ti -1c
.RI "std::string \fBgetImage\fP ()"
.br
.ti -1c
.RI "int \fBgetImageHeight\fP ()"
.br
.ti -1c
.RI "int \fBgetImageWidth\fP ()"
.br
.ti -1c
.RI "int \fBgetMargin\fP ()"
.br
.ti -1c
.RI "std::string \fBgetName\fP ()"
.br
.ti -1c
.RI "int \fBgetSpacing\fP ()"
.br
.ti -1c
.RI "int \fBgetTileCount\fP ()"
.br
.ti -1c
.RI "std::string \fBgetTiledVersion\fP ()"
.br
.ti -1c
.RI "int \fBgetTileHeight\fP ()"
.br
.ti -1c
.RI "int \fBgetTileWidth\fP ()"
.br
.ti -1c
.RI "std::string \fBgetType\fP ()"
.br
.ti -1c
.RI "float \fBgetVersion\fP ()"
.br
.in -1c
.SS "Private Attributes"

.in +1c
.ti -1c
.RI "int \fBfirstgid\fP"
.br
.RI "Zmienna przechowująca numer kafelka\&. "
.ti -1c
.RI "int \fBcolumns\fP"
.br
.RI "Zmienna przechowujaca numer kolumny\&. "
.ti -1c
.RI "std::string \fBimage\fP"
.br
.RI "Zmienna przechowujaca sciezke do obrazka dla kafelka\&. "
.ti -1c
.RI "int \fBimageHeight\fP"
.br
.RI "Zmienna przechowujaca wysokosc obrazka kafelka\&. "
.ti -1c
.RI "int \fBimageWidth\fP"
.br
.RI "Zmienna przechowujaca szerokosc obrazka kafelka\&. "
.ti -1c
.RI "int \fBmargin\fP"
.br
.RI "Zmienna przechowujaca margines\&. "
.ti -1c
.RI "std::string \fBname\fP"
.br
.RI "Zmienna przechowujaca nazwe\&. "
.ti -1c
.RI "int \fBspacing\fP"
.br
.RI "Zmienna przechowujaca odstep\&. "
.ti -1c
.RI "int \fBtilecount\fP"
.br
.RI "Zmienna przechowuajca ilosc kafelkow\&. "
.ti -1c
.RI "std::string \fBtiledversion\fP"
.br
.RI "Zmienna przechowujaca informacje o wersji kafelka\&. "
.ti -1c
.RI "int \fBtileheight\fP"
.br
.RI "Zmienna przechowujaca wysokosc kafelka\&. "
.ti -1c
.RI "int \fBtilewidth\fP"
.br
.RI "Zmienna przechowujaca szerokosc kafelka\&. "
.ti -1c
.RI "std::string \fBtype\fP"
.br
.RI "Zmienna przechowujaca inofrmacje o typie\&. "
.ti -1c
.RI "float \fBversion\fP"
.br
.RI "Zmienna przechowujca inforamcje o wersji na ktorej zostal stworzony kafelek\&. "
.in -1c
.SH "Constructor & Destructor Documentation"
.PP 
.SS "TileSet::TileSet ()\fC [inline]\fP"

.PP
Konstruktor pusty\&. 
.SS "TileSet::TileSet (int firstgid, std::ifstream tileset)"
Konstruktor który przyjmuje numer kafelka i jego tileset a nastepnie go parsuje 
.PP
\fBParameters\fP
.RS 4
\fIfirstgid\fP - pierwszy globalny identyfikator kafelka tego zestawu kafelków 
.br
\fItileset\fP - zestaw kafelków 
.RE
.PP

.SS "TileSet::~TileSet ()\fC [default]\fP"

.PP
Destruktor\&. 
.SH "Member Function Documentation"
.PP 
.SS "int TileSet::getColumns ()\fC [inline]\fP"
Zwraca numer kolumny 
.PP
\fBReturns\fP
.RS 4
zwraca liczbe, która reprezentuje liczbe kolumn kafelków w danym zestawie 
.RE
.PP

.SS "int TileSet::getFirstGid ()\fC [inline]\fP"
Metoda która zwraca informacje o pierwszy globalny identyfikator kafelka wybranego zestawu 
.PP
\fBReturns\fP
.RS 4
zwraca liczbe która reprezentuje id pierwszego kafelka w danym zestawie 
.RE
.PP

.SS "std::string TileSet::getImage ()\fC [inline]\fP"
Zwraca informacje o obrazku kafelka 
.PP
\fBReturns\fP
.RS 4
zwraca tekst, który reprezentuje ścieżke do obrazka dla kafelka 
.RE
.PP

.SS "int TileSet::getImageHeight ()\fC [inline]\fP"
Zwraca wysokosc obrazka kafelka 
.PP
\fBReturns\fP
.RS 4
zwraca liczbe, która reprezentuje wysokość w pikselach dla obrazka kafelka 
.RE
.PP

.SS "int TileSet::getImageWidth ()\fC [inline]\fP"
Zwraca szerokosc obrazka kafelka 
.PP
\fBReturns\fP
.RS 4
zwraca liczbe, która reprezentuje szerokosc w pikselach dla obrazka kafelka 
.RE
.PP

.SS "int TileSet::getMargin ()\fC [inline]\fP"
Zwraca margines 
.PP
\fBReturns\fP
.RS 4
zwraca liczbe, która reprezentuje margines w tym zestawie kafelków 
.RE
.PP

.SS "std::string TileSet::getName ()\fC [inline]\fP"
Zwraca nazwe 
.PP
\fBReturns\fP
.RS 4
zwraca tekst, który reprezentuje nazwe tego zestawu kafelka 
.RE
.PP

.SS "int TileSet::getSpacing ()\fC [inline]\fP"
Zwraca odstep 
.PP
\fBReturns\fP
.RS 4
zwraca liczbę, która reprezentuje odstęp w pikselach między kafelkami w tym zestawie 
.RE
.PP

.SS "int TileSet::getTileCount ()\fC [inline]\fP"
Zwraca ilosc kafelkow 
.PP
\fBReturns\fP
.RS 4
zwraca liczbe, która reprezentuje ilość kafelków w tym zestawie 
.RE
.PP

.SS "std::string TileSet::getTiledVersion ()\fC [inline]\fP"
Zwraca informacje o wersji kafelka 
.PP
\fBReturns\fP
.RS 4
zwraca tekst, który reprezentuje wersję w której został zapisany kafelek 
.RE
.PP

.SS "int TileSet::getTileHeight ()\fC [inline]\fP"
Zwraca wysokosc kafelka 
.PP
\fBReturns\fP
.RS 4
zwraca liczba która reprezentuje wysokość kafelka 
.RE
.PP

.SS "int TileSet::getTileWidth ()\fC [inline]\fP"
Zwraca szerokosc kafelka 
.PP
\fBReturns\fP
.RS 4
zwraca liczbę która reprezentuje szerokość kafelka 
.RE
.PP

.SS "std::string TileSet::getType ()\fC [inline]\fP"
Zwraca typ 
.PP
\fBReturns\fP
.RS 4
zwraca tekst, który reprezentuje typ 
.RE
.PP

.SS "float TileSet::getVersion ()\fC [inline]\fP"
Zwraca numer wersji na której został stworzony kafelek 
.PP
\fBReturns\fP
.RS 4
zwraca liczbę float, która reprezentuje wersję w której został stworzony kafelek 
.RE
.PP

.SH "Member Data Documentation"
.PP 
.SS "int TileSet::columns\fC [private]\fP"

.PP
Zmienna przechowujaca numer kolumny\&. 
.SS "int TileSet::firstgid\fC [private]\fP"

.PP
Zmienna przechowująca numer kafelka\&. 
.SS "std::string TileSet::image\fC [private]\fP"

.PP
Zmienna przechowujaca sciezke do obrazka dla kafelka\&. 
.SS "int TileSet::imageHeight\fC [private]\fP"

.PP
Zmienna przechowujaca wysokosc obrazka kafelka\&. 
.SS "int TileSet::imageWidth\fC [private]\fP"

.PP
Zmienna przechowujaca szerokosc obrazka kafelka\&. 
.SS "int TileSet::margin\fC [private]\fP"

.PP
Zmienna przechowujaca margines\&. 
.SS "std::string TileSet::name\fC [private]\fP"

.PP
Zmienna przechowujaca nazwe\&. 
.SS "int TileSet::spacing\fC [private]\fP"

.PP
Zmienna przechowujaca odstep\&. 
.SS "int TileSet::tilecount\fC [private]\fP"

.PP
Zmienna przechowuajca ilosc kafelkow\&. 
.SS "std::string TileSet::tiledversion\fC [private]\fP"

.PP
Zmienna przechowujaca informacje o wersji kafelka\&. 
.SS "int TileSet::tileheight\fC [private]\fP"

.PP
Zmienna przechowujaca wysokosc kafelka\&. 
.SS "int TileSet::tilewidth\fC [private]\fP"

.PP
Zmienna przechowujaca szerokosc kafelka\&. 
.SS "std::string TileSet::type\fC [private]\fP"

.PP
Zmienna przechowujaca inofrmacje o typie\&. 
.SS "float TileSet::version\fC [private]\fP"

.PP
Zmienna przechowujca inforamcje o wersji na ktorej zostal stworzony kafelek\&. 

.SH "Author"
.PP 
Generated automatically by Doxygen for Neon Jumper from the source code\&.


src/TileSet.cpp
#include <fstream>
#include <iostream>
#include "TileSet.h"
#include "rapidjson/include/rapidjson/ostreamwrapper.h"
#include "rapidjson/include/rapidjson/document.h"
#include "rapidjson/include/rapidjson/istreamwrapper.h"
#include "rapidjson/include/rapidjson/writer.h"
#include "rapidjson/include/rapidjson/stringbuffer.h"
#include "rapidjson/include/rapidjson/ostreamwrapper.h"

TileSet::TileSet(int firstgid, std::ifstream tileset) {
    this->firstgid = firstgid;
    rapidjson::Document tilesetDoc;
    rapidjson::IStreamWrapper isw {tileset};
    tilesetDoc.ParseStream(isw);

    columns = tilesetDoc["columns"].GetInt();
    image = tilesetDoc["image"].GetString();
    imageHeight = tilesetDoc["imageheight"].GetInt();
    imageWidth = tilesetDoc["imagewidth"].GetInt();
    margin = tilesetDoc["margin"].GetInt();
    name = tilesetDoc["name"].GetString();
    spacing = tilesetDoc["spacing"].GetInt();
    tilecount = tilesetDoc["tilecount"].GetInt();
    tiledversion = tilesetDoc["tiledversion"].GetString();
    tileheight = tilesetDoc["tileheight"].GetInt();
    tilewidth = tilesetDoc["tilewidth"].GetInt();
    type = tilesetDoc["type"].GetString();
    version = tilesetDoc["version"].GetFloat();
    tileset.close();
    //delete[](tileset);
}

TileSet::~TileSet() = default;
